optuna/trial.py

```diff
@@ -39,6 +39,11 @@ class TrialState(enum.Enum):
             :class:`~optuna.exceptions.TrialPruned`.
         FAIL:
             The :class:`~optuna.trial.Trial` has failed due to an uncaught error.
+
+            ... note::
+                The failed trials are ignored by any build-in samplers when they sample new
+                parameters. Thus, failed trials are regarded as deleted in the samplers'
+                perspective.
```

Comment:
How about adding the information in `BaseSampler` description, because uses of samplers want to know it.

https://optuna.readthedocs.io/en/stable/reference/samplers.html (Sorry, I do not know exactly where to add it in the page...)

Reply:
Thank you for your comment! I will add the information in `sampler_independent` and `sampler_relative`.